orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/StartExecutionHandler.kt

```diff
@@ -34,15 +36,20 @@ class StartExecutionHandler(
 
   override fun handle(message: StartExecution) {
     message.withExecution { execution ->
-      repository.updateStatus(message.executionId, RUNNING)
+      // isCanceled flag lets the pipeline clean itself up, and may be true before the status is set to CANCELED
+      if (execution.status != NOT_STARTED || execution.isCanceled) {
+        publisher.publishEvent(ExecutionComplete(this, message.executionType, message.executionId, execution.status))
```

Comment:
Do we really want to fire this message for _any_ other status? Trying to think what the possible causes are here.

Reply:
Changing as we discussed - not sending message if it's not canceled.